chore(docs): note cache-control header for preview/draft mode (#53825)
vercel/next.js: docs/03-pages/01-building-your-application/06-configuring/11-draft-mode.mdx

### Works with `getServerSideProps`

> Draft Mode works with `getServerSideProps` as well. It will also be available on the `context` object containing `draftMode`.

Reviewer: Small nitpick, I know you didn't write this bit :) 

```suggestion
Draft Mode works with `getServerSideProps`, and is available as a key in the [`context`](/docs/pages/api-reference/functions/get-server-side-props#context-parameter) object.
```

> > **Good to know**: Avoid setting the `Cache-Control` header because it cannot be bypassed. Instead, use [ISR](/docs/pages/building-your-application/data-fetching/incremental-static-regeneration).

Reviewer 2: I don't think we need to recommend avoiding setting it as it is valid to use still just not in combination with draft mode.

Reviewer: Suggestion: 

```suggestion
> **Good to know**: You shouldn't set the `Cache-Control` header when using Draft Mode because it cannot be bypassed. Instead, we recommend using [ISR](/docs/pages/building-your-application/data-fetching/incremental-static-regeneration).
```
